src/forms/TimeInput.jsx

```diff
@@ -54,29 +152,102 @@ class TimeInput extends React.Component {
 		}
 
 		return (
-			<span>
+			<div>
 				{ label && <label htmlFor={id} className={labelClassNames}>{label}</label> }
-				<input
-					id={id}
-					type='time'
-					name={name}
-					value={value}
-					className={classNames}
-					required={required}
-					onChange={this.onChange}
-					ref={ input => this.inputEl = input }
-					{...other}
-				/>
+				{
+					this.state.supportsTime
+					?
+						<input
+							id={id}
+							type='time'
+							name={name}
+							value={value}
+							className={classNames}
+							required={required}
+							disabled={disabled}
+							onChange={this.onChange}
+							ref={ input => this.inputEl = input }
+							{...other}
+						/>
+					:
+						<div>
+							<div className={fauxInputClassNames}>
+								<Flex>
+									<FlexItem>
+										<input type="text"
+											pattern="\d*"
+											id="hours"
+											name="hours"
+											min={is24Hr ? 0 : 1}
+											max={is24Hr ? 23 : 12}
+											disabled={disabled}
+											className={`field--reset align--center ${HOURS_INPUT_CLASS}`}
+											onBlur={this.onBlur}
+											onChange={this.onNumberChange}
+											maxLength={2}
+											value={this.state.hours} /> {/* is24Hr ? this.state.hours % 24 : this.state.hours % 12 */}
+									</FlexItem>
+									<FlexItem className="align--center">
+										{':'}
+									</FlexItem>
+									<FlexItem>
+										<input type="text"
+											pattern="\d*"
+											id="minutes"
+											name="minutes"
+											min={0}
+											max={59}
+											disabled={disabled}
+											className={`field--reset align--center ${MINUTES_INPUT_CLASS}`}
+											onBlur={this.onBlur}
+											onChange={this.onNumberChange}
+											maxLength={2}
+											value={this.state.minutes} />
+									</FlexItem>
+									{ !is24Hr &&
+										<FlexItem shrink className="timeInput-meridianContainer">
+											<SelectInput
+												id="meridian"
+												name="meridian"
+												className={meridianClassNames}
+												disabled={disabled}
+												onChange={this.onMeridianChange}
+												value={this.state.meridian}
+												options={[
+													{ label: 'AM', value: 'AM' },
+													{ label: 'PM', value: 'PM' }
+												]}
+											/>
+										</FlexItem>
+									}
+								</Flex>
+							</div>
+							<input
+								type="hidden"
+								className={HIDDEN_INPUT_CLASS}
+								id={id}
+								name={name}
+								value={`${formatHours(this.state.hours, this.state.meridian)}:${this.state.minutes}`}
+								onChange={this.onChange}
```

Comment:
so I don't think `onChange` fires with a hidden input, there needs to be another way to tell redux-form that the values have changed and fire `this.props.onChnage`